### ISSUE TITLE
Extension of `ignite.distributed.utils.broadcast` for pathlib.Path datatype

### ISSUE THREAD

**Issue by guptaaryan16** [pytorch/ignite#3025]
## 🚀 Feature

I would like to request for extension of `ignite.distributed.utils.broadcast` for Path datatype as it is frequently used for distributed training and can be very useful for designing DL pipeline based projects. The API should work like this 

```
from pathlib import Path
import ignite.distributed as idist

path = Path("./log")

idist.broadcast(path, src=0)
```
Further if possible we can also provide support for more data types as suggested by others. Also if possible I would like to take up this issue.
@vfdev-5 @sadra-barikbin

**Comment by guptaaryan16** [comment]
@sadra-barikbin Yeah it does but I was hoping more data types can be supported as it seems possible in the codebase

**Comment by nightcityblade** [comment]
Hi, I'd like to work on this. I'll submit a PR shortly.
